pkg/proxy/endpoints_test.go

```diff
@@ -400,7 +400,7 @@ func TestEndpointsToEndpointsMap(t *testing.T) {
 			} else {
 				for i := range newEndpoints[x] {
 					ep := newEndpoints[x][i].(*BaseEndpointInfo)
-					if *ep != *(tc.expected[x][i]) {
+					if !(reflect.DeepEqual(*ep, *(tc.expected[x][i]))) {
```

Comment:
`Topology` embedded into `BaseEndpointInfo`, which is a map, struct contains map cannot use `!=` operator, so change it to `reflect.DeepEqual`.